lib/core/src/sdk.rs

```diff
@@ -741,6 +741,20 @@ impl LiquidSdk {
             match self.persister.get_payment(&id)? {
                 Some(payment) => {
                     self.update_wallet_info().await?;
+                    
+                    // For Complete status, re-fetch the payment to ensure we have the most up-to-date data
+                    // after the wallet_info update, which may have updated balances
+                    let payment = match payment.status {
+                        Complete => {
+                            // Try to get the refreshed payment data
+                            match self.persister.get_payment(&id)? {
+                                Some(refreshed_payment) => refreshed_payment,
+                                None => payment // Fall back to original payment data if not found
+                            }
+                        },
+                        _ => payment
+                    };
```

Comment:
We already fetch the payment above. To save the extra fetch (and code) perhaps it is better to move the call to update_wallet_info right before the first fetch?